beacon-chain/p2p/custody.go

```diff
@@ -9,18 +12,14 @@ import (
 	"github.com/prysmaticlabs/prysm/v5/config/params"
 )
 
+var _ DataColumnsHandler = (*Service)(nil)
+
 // AdmissibleCustodyGroupsPeers returns a list of peers that custody a super set of the local node's custody groups.
 func (s *Service) AdmissibleCustodyGroupsPeers(peers []peer.ID) ([]peer.ID, error) {
```

Comment:
This function should not be used at all.
Why is the issue when trying to get rid of it?

Reply:
The new logic was initially tied to the pending blocks queue service but the remaining call to this function was in the initial sync. I've extracted the logic into functions that can be used in both places and removed this function.